### PR TITLE
php-formula: added missing dependencies, removed DTrace & LDAP SALS extension.

### PR DESCRIPTION
**Pull request by zgjonbalaj**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [-] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

I'm getting an error on audit which I believe has to do with another issue.

```
➜  homebrew-core git:(php-formula) brew audit --strict php
==> Installing or updating 'rubocop' gem
Error: Unable to resolve dependency: user requested 'did_you_mean (= 1.2.0)'
➜  homebrew-core git:(php-formula) 

``` 

-----

### CODE REVIEW

**Comment by zgjonbalaj** [comment]
After looking at this, I think the proper thing to do is do a check on the --with-dtrace flag to check if its a OS.Linux.

File: Formula/php.rb [hunk 1]
```diff
@@ -25,14 +27,18 @@ class Php < Formula
...
   depends_on "libpq"
   depends_on "libsodium"
+  depends_on "libxslt"
   depends_on "libzip"
+  depends_on "openldap"
   depends_on "openssl"
   depends_on "pcre"
   depends_on "unixodbc"
   depends_on "webp"
+  depends_on "zlib"
```
Comment (iMichka) [inline Formula/php.rb]: Please regroup all new linux-only dependencies in a single block, safeguarded by `unless OS.mac? .... end`.

File: Formula/php.rb [hunk 2]
```diff
@@ -138,8 +144,8 @@ def install
       --with-unixODBC=#{Formula["unixodbc"].opt_prefix}
       --with-webp-dir=#{Formula["webp"].opt_prefix}
       --with-xmlrpc
-      --with-xsl
-      --with-zlib
+      --with-xsl-dir=#{Formula["libxslt"].opt_prefix}
+      --with-zlib-dir=#{Formula["zlib"].opt_prefix}
```
Comment (iMichka) [inline Formula/php.rb]: Please regroup the flags at the end like this:
```
if OS.mac?
   args << "-mac args"
else
  args << "-linux args"
end

File: Formula/php.rb [hunk 3]
```diff
@@ -389,4 +395,4 @@ def plist; <<~EOS
       end
     end
   end
-end
+end
```
Comment (iMichka) [inline Formula/php.rb]: Do not remove the carriage return here.

**Comment by iMichka** [comment]
Ping @zgjonbalaj

**Pull request closed without merging**